webapp/sources/rudder/rudder-rest/src/test/scala/com/normation/rudder/MockServices.scala

```diff
@@ -3031,10 +3032,23 @@ class MockCampaign() {
   object repo extends CampaignRepository {
     val items = Ref.make(Map[CampaignId, DumbCampaignTrait]((c0.info.id -> c0))).runNow
 
-    override def getAll():            IOResult[List[Campaign]] = items.get.map(_.valuesIterator.toList)
-    override def get(id: CampaignId): IOResult[Campaign]       =
+    override def getAll(typeFilter: List[CampaignType], statusFilter: List[CampaignStatusValue]): IOResult[List[Campaign]] = {
+      for {
+        campaigns <- items.get.map(_.valuesIterator.toList)
+      } yield {
+        (typeFilter, statusFilter) match {
+          case (Nil, Nil) => campaigns
+          case (t, Nil)   => campaigns.filter(c => t.contains(c.campaignType))
+          case (Nil, s)   => campaigns.filter(c => s.contains(c.info.status.value))
+          case (t, s)     => campaigns.filter(c => t.contains(c.campaignType) && s.contains(c.info.status.value))
+        }
```

Comment:
you already have that exact code in `CampaignRepository`, it likely means that that bit of logic needs to be refactored out in an utility object (`object CampaignRepository` likely) and reused in both places.